src/SaveObject.ts

```diff
@@ -864,6 +885,63 @@ function download(filename: string, content: string): void {
   }, 0);
 }
 
+async function compress(str: string): Promise<Uint8Array> {
+  const stream = new Blob([str]).stream();
+  const compressedStream = stream.pipeThrough(new CompressionStream("gzip"));
+  const chunks = [];
+  const reader = compressedStream.getReader();
+  try {
+    let done = false; // to make lint shut up
+    while (!done) {
+      const v = await reader.read();
+      if ((done = v.done)) break;
+      chunks.push(v.value);
+    }
+  } finally {
+    reader.releaseLock();
+  }
+  return await concatUint8Arrays(chunks);
+}
+
+async function decompress(compressedBytes: Uint8Array): Promise<string> {
+  const stream = new Blob([compressedBytes]).stream();
+  const decompressedStream = stream.pipeThrough(new DecompressionStream("gzip"));
+  const chunks = [];
+  const reader = decompressedStream.getReader();
+  try {
+    let done = false;
+    while (!done) {
+      const v = await reader.read();
+      if ((done = v.done)) break;
+      chunks.push(v.value);
+    }
+  } finally {
+    reader.releaseLock();
+  }
+  const stringBytes = await concatUint8Arrays(chunks);
+
+  // Convert the bytes to a string.
+  return new TextDecoder().decode(stringBytes);
+}
+
+async function concatUint8Arrays(uint8arrays: Uint8Array[]): Promise<Uint8Array> {
+  const blob = new Blob(uint8arrays);
+  const buffer = await blob.arrayBuffer();
+  return new Uint8Array(buffer);
+}
+
+// This is not the most efficient implementation;
+// See https://developer.mozilla.org/en-US/docs/Glossary/Base64
```

Comment:
I would actually suggest using the gross but efficient implementations. If not that, then the better versions of these that don't use closures. (The difference is significant.)